text/XXXX-config_revision.md

```diff
@@ -0,0 +1,268 @@
+- Feature Name: configuration hierarchy
+- Start Date: <!-- (fill me in with today's date, YYYY-MM-DD) -->
+- RFC PR: <!-- (leave this empty) -->
+- Issue: <!-- (leave this empty) -->
+
+# Summary
+[summary]: #summary
+
+Define a hierarchy for configuration overrides, revise configuration fields with clear descriptions.
+
+# Motivation
+[motivation]: #motivation
+
+Our configuration story needs work. _Many_ fields are not in use, and there's ambiguity about how 
+
+This RFC proposes a few concrete changes:
+1. Overhaul configuration data structures, removing unnecessary fields & adding new ones
+1. Define how environment variables and configuration interact.
+1. Formalize the hierarchy of different configuration sources
+1. Formalize the list of enviornment variables Qri is sensitive to.
+1. Switch to multiaddresses for all network configuration.
+1. Move ipfs repo location into configuration, deprecate support for `IPFS_PATH`.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+
+### Environment Variables don't configure, they point to configuration files
+
+This is the total list of environemnt variables qri is sensitive to:
+
+| variable name            | description                           |
+| ------------------------ | ------------------------------------- |
+| `PAGER`                  | what pager to use for output          |
+| `QRI_PATH`               | path to qri configuraiton data        |
+| `QRI_SETUP_CONFIG_DATA`  | JSON data to use when creating a repo |
+| `IPFS_SETUP_CONFIG_DATA` | JSON data when initializing IPFS      |
+| `QRI_PRIVATE_KEY`        | defined in the private key RFC        |
+| `QRI_BACKTRACE`          | show stack trace output on crash      |
+
+
+_note: I (b5) haven't had a chance to vet our dependencies for env vars they are sensitive to, so this list is only complete for the purpose of this RFC._
+
+This RFC proposes environment variables are _not_ a direct source of configuration. Instead environment variables either point to a path to load configuration from, or contain a payload to create an initial configuration file.
+
+`QRI_SETUP_CONFIG_DATA` as the only supported is only honored by `qri setup`. We need `SETUP` env variables for operating within containerized contexts, but that's about it.
```

Comment:
A bit confused about what the first sentence is saying here.

"`QRI_SETUP_CONFIG_DATA`  is only honored by `qri setup`." ??

Reply:
meaning this variable is only _used_ when running `qri setup`. All other times, it's value is ignored